common/src/main/java/dev/isxander/yacl3/gui/controllers/dropdown/AbstractDropdownControllerElement.java

```diff
@@ -0,0 +1,193 @@
+package dev.isxander.yacl3.gui.controllers.dropdown;
+
+import com.mojang.blaze3d.platform.InputConstants;
+import com.mojang.blaze3d.vertex.PoseStack;
+import dev.isxander.yacl3.api.utils.Dimension;
+import dev.isxander.yacl3.gui.YACLScreen;
+import dev.isxander.yacl3.gui.controllers.string.StringControllerElement;
+import dev.isxander.yacl3.gui.utils.GuiUtils;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.Component;
+
+import java.util.List;
+
+public abstract class AbstractDropdownControllerElement<T, U> extends StringControllerElement {
+	public static final int MAX_SHOWN_NUMBER_OF_ITEMS = 7;
+	private final AbstractDropdownController<T> dropdownController;
+	protected boolean opened = false;
+	// Stores the current selection position. The item at this position in the dropdown list will be chosen as the
+	// accepted value when the element is closed.
+	protected int selected = 0;
+
+	public AbstractDropdownControllerElement(AbstractDropdownController<T> control, YACLScreen screen, Dimension<Integer> dim) {
+		super(control, screen, dim, false);
+		this.dropdownController = control;
+	}
+
+	public void open() {
```

Comment:
Could this be renamed to something more clear, like `showDropdown()`? Same with close.